File: Serene/Serene.Script/Northwind/Order/OrderGrid.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System.Runtime.CompilerServices;

    [Imported]
    public class OrderGrid : EntityGrid<OrderRow>
    {
        public OrderGrid(jQueryObject container)
            : base(container)
        {
        }

        [IntrinsicProperty]
        public CustomerEditor CustomerFilter { get; private set; }
    }

    /* 
    This class has been ported to TypeScript. See OrderGrid.ts
    Code below is only a reference for those who want to use Saltaralle

    [ColumnsKey("Northwind.Order"), IdProperty("OrderID")]
    [DialogType(typeof(OrderDialog)), LocalTextPrefix("Northwind.Order"), Service("Northwind/Order")]
    public class OrderGrid : EntityGrid<OrderRow>
    {
        private EnumEditor shippingState;

        public OrderGrid(jQueryObject container)
            : base(container)
        {
        }

        protected override void CreateToolbarExtensions()
        {
            base.CreateToolbarExtensions();

            CustomerFilter = AddEqualityFilter<CustomerEditor>(Fields.CustomerID);

            AddDateRangeFilter(Fields.OrderDate);

            shippingState = AddEqualityFilter<EnumEditor>(Fields.ShippingState,
                options: new EnumEditorOptions { EnumKey = "Northwind.OrderShippingState" });

            AddEqualityFilter<LookupEditor>(Fields.ShipVia,
                options: new LookupEditorOptions { LookupKey = ShipperRow.LookupKey });

            AddEqualityFilter<LookupEditor>(Fields.ShipCountry,
                options: new LookupEditorOptions { LookupKey = "Northwind.OrderShipCountry" });

            AddEqualityFilter<LookupEditor>(Fields.ShipCity, options: new LookupEditorOptions {
                LookupKey = "Northwind.OrderShipCity", CascadeFrom = Fields.ShipCountry });

            AddEqualityFilter<LookupEditor>(Fields.EmployeeID,
                options: new LookupEditorOptions { LookupKey = EmployeeRow.LookupKey });
        }

        public OrderShippingState? ShippingState
        {
            get { return (OrderShippingState?)shippingState.Value.ToInt32(); }
            set { shippingState.Value = value == null ? "" : ((int)value).ToString(); }
        }

        protected override List<ToolButton> GetButtons()
        {
            var buttons = base.GetButtons();

            buttons.Add(Common.ExcelExportHelper.CreateToolButton(this,
                OrderService.BaseUrl + "/ListExcel", this.OnViewSubmit));

            buttons.Add(Common.PdfExportHelper.CreateToolButton(this, this.OnViewSubmit));

            return buttons;
        }

        [IntrinsicProperty]
        public CustomerEditor CustomerFilter { get; private set; }
    }
    */
}

File: Serene/Serene.Web/Modules/Northwind/Order/OrderGrid.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.filterable()
    export class OrderGrid extends Serenity.EntityGrid<OrderRow, any> {
        protected getColumnsKey() { return "Northwind.Order"; }
        protected getDialogType() { return <any>OrderDialog; }
        protected getIdProperty() { return OrderRow.idProperty; }
        protected getLocalTextPrefix() { return OrderRow.localTextPrefix; }
        protected getService() { return OrderService.baseUrl; }

        protected shippingStateFilter: Serenity.EnumEditor;

        constructor(container: JQuery) {
            super(container);
        }

        protected createQuickFilters() {
            super.createQuickFilters();

            let fld = OrderRow.Fields;
            this.shippingStateFilter = this.findQuickFilter(Serenity.EnumEditor, fld.ShippingState);
        }

        protected getButtons()
        {
            var buttons = super.getButtons();

            buttons.push(Common.ExcelExportHelper.createToolButton({
                grid: this,
                service: OrderService.baseUrl + '/ListExcel',
                onViewSubmit: () => this.onViewSubmit(),
                separator: true
            }));

            buttons.push(Common.PdfExportHelper.createToolButton({
                grid: this,
                onViewSubmit: () => this.onViewSubmit()
            }));

            return buttons;
        }

        public set_shippingState(value: number): void {
            this.shippingStateFilter.value = value == null ? '' : value.toString();
        }
    }
}
